python/lsst/daf/butler/_butler.py

```diff
@@ -1770,3 +1803,69 @@ def prepend_collection_chain(
         transactions short.
         """
         raise NotImplementedError()
+
+    @abstractmethod
+    def extend_collection_chain(
+        self, parent_collection_name: str, child_collection_names: str | Iterable[str]
+    ) -> None:
+        """Add children to the end of a CHAINED collection.
+
+        If any of the children already existed in the chain, they will be moved
```

Comment:
Moving existing children is less obviously useful/correct than it is for `prepend`.  Not sure what would be better though, and this is at least consistent.